homeassistant/config_entries.py

```diff
@@ -2726,6 +2726,16 @@ def is_matching(self, other_flow: Self) -> bool:
         """Return True if other_flow is matching this flow."""
         raise NotImplementedError
 
+    @callback
+    def _get_entry_from_context(self) -> ConfigEntry:
+        """Return current config entry, based on context['entry_id']."""
```

Comment:
This only works for reauth and reconfigure flows, right?

I think that should be clear from the docstring.
Maybe we should check the source of the flow is either SOURCE_REAUTH or SOURCE_RECONFIGURE?

Also, can the method name better reflect when this method can be used, instead of the implementation detail which is where we store the entry id?
Maybe two methods, `_get_reauth_entry` + `get_reconfigure_entry`?

Reply:
If I'm not mistaken it could also be used in options flow.
I'll double check when I'm back on my PC.

Reply:
I have renamed the method, and adjusted the docstring to make it clearer when it can be used.

I don't think checking the source is necessary.

I think having two methods doing the exact same thing would simply add confusion...